nitro-node/src/nitro.ts

```diff
@@ -0,0 +1,511 @@
+import os from "node:os";
+import fs from "node:fs";
+import path from "node:path";
+import { ChildProcessWithoutNullStreams, spawn } from "node:child_process";
+import tcpPortUsed from "tcp-port-used";
+import fetchRT from "fetch-retry";
+import osUtils from "os-utils";
+import {
+  getNitroProcessInfo,
+  updateNvidiaInfo as _updateNvidiaInfo,
+} from "./nvidia";
+import { executableNitroFile } from "./execute";
+import {
+  NitroNvidiaConfig,
+  NitroModelSetting,
+  NitroPromptSetting,
+  NitroLogger,
+  NitroModelOperationResponse,
+  NitroModelInitOptions,
+  ResourcesInfo,
+} from "./types";
+import { downloadNitro } from "./scripts";
+// Polyfill fetch with retry
+const fetchRetry = fetchRT(fetch);
+
+// The PORT to use for the Nitro subprocess
+const PORT = 3928;
+// The HOST address to use for the Nitro subprocess
+const LOCAL_HOST = "127.0.0.1";
+// The URL for the Nitro subprocess
+const NITRO_HTTP_SERVER_URL = `http://${LOCAL_HOST}:${PORT}`;
+// The URL for the Nitro subprocess to load a model
+const NITRO_HTTP_LOAD_MODEL_URL = `${NITRO_HTTP_SERVER_URL}/inferences/llamacpp/loadmodel`;
+// The URL for the Nitro subprocess to validate a model
+const NITRO_HTTP_VALIDATE_MODEL_URL = `${NITRO_HTTP_SERVER_URL}/inferences/llamacpp/modelstatus`;
+// The URL for the Nitro subprocess to kill itself
+const NITRO_HTTP_KILL_URL = `${NITRO_HTTP_SERVER_URL}/processmanager/destroy`;
+// The URL for the Nitro subprocess to run chat completion
+const NITRO_HTTP_CHAT_URL = `${NITRO_HTTP_SERVER_URL}/inferences/llamacpp/chat_completion`;
+
+// The default config for using Nvidia GPU
+const NVIDIA_DEFAULT_CONFIG: NitroNvidiaConfig = {
+  notify: true,
+  run_mode: "cpu",
```

Comment:
These are not related to nitro-node and GPU.

Reply:
Can consider run_mode, but should not be overwritten. Can set default "gpu" when its not defined and CUDA check passed. Otherwise app can overwrite (nitro extension should cache it somewhere, e.g. settings.json before)

Reply:
Oops, missed this detail. Will refactor the logic for this, this field should be an optional parameter to be specified at runtime, not in the runtime-persistent config passed around globally.